dependency/vault_read_test.go

```diff
@@ -325,6 +326,34 @@ func TestVaultReadQuery_Fetch_KVv2(t *testing.T) {
 		})
 	}
 
+	t.Run("read_deleted", func(t *testing.T) {
+		// only needed for KVv2 as KVv1 doesn't have metadata
+		path := "data/foo/zed"
+		// create and delete a secret
+		err = vault.CreateSecret(path, map[string]interface{}{
+			"zip": "zop",
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+		err = vault.deleteSecret(path)
+		if err != nil {
+			t.Fatal(err)
+		}
+		// now look for entry with metadata but no data (deleted secret)
+		path = vault.secretsPath + "/" + path
+		d, err := NewVaultReadQuery(path)
+		if err != nil {
+			t.Fatal(err)
+		}
+		_, _, err = d.Fetch(clients, nil)
+		if err == nil {
+			t.Fatal("Nil received when error expected")
+		}
+		exp_err := fmt.Errorf("no secret exists at %s", path)
+		assert.Equal(t, exp_err, errors.Cause(err))
```

Comment:
regardless of whether the team prefers using `assert` or `t.Fatal` for tests, we should at least keep it consistent within a test. I'd like to see here instead, `if exp_err != errors.Cause(err); t.Fatal` or else a change in the rest of the test assertions. wdyt?

Reply:
Yeah... I was going back and forth about using assert here and ended up in between. Assert is used throughout the tests and so I thought staying consistent would be good, but being more idiomatic would also be good.

I'm going to try the more idiomatic approach and see how it works going forward.

Reply:
Ended up with a string comparison for the error compare. Hope the 1.13 errors changes help with this.